uber/models/department.py

```diff
@@ -0,0 +1,387 @@
+import uuid
+from datetime import timedelta
+
+import six
+from sideboard.lib import cached_property
+from sideboard.lib.sa import CoerceUTF8 as UnicodeText, \
+    UTCDateTime, UUID
+from sqlalchemy import and_, or_, exists, select
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import backref, column_property
+from sqlalchemy.schema import ForeignKey, Table, UniqueConstraint
+from sqlalchemy.types import Boolean, Float, Integer
+
+from uber.config import c
+from uber.decorators import classproperty
+from uber.models import MagModel
+from uber.models.attendee import Attendee
+from uber.models.types import default_relationship as relationship, \
+    Choice, DefaultColumn as Column
+from uber.utils import comma_and
+
+
+__all__ = [
+    'dept_membership_dept_role', 'job_required_role', 'Department',
+    'DeptChecklistItem', 'DeptMembership', 'DeptMembershipRequest',
+    'DeptRole', 'Job', 'Shift']
+
+
+# Many to many association table to represent the DeptRoles fulfilled
+# by a DeptMembership
+dept_membership_dept_role = Table(
+    'dept_membership_dept_role',
+    MagModel.metadata,
+    Column('dept_membership_id', UUID, ForeignKey('dept_membership.id')),
+    Column('dept_role_id', UUID, ForeignKey('dept_role.id')))
+
+
+# Many to many association table to represent the DeptRoles required
+# to fulfill a Job
+job_required_role = Table(
+    'job_required_role',
+    MagModel.metadata,
+    Column('job_id', UUID, ForeignKey('job.id')),
+    Column('dept_role_id', UUID, ForeignKey('dept_role.id')))
+
+
+class DeptChecklistItem(MagModel):
+    department_id = Column(UUID, ForeignKey('department.id'))
+    attendee_id = Column(UUID, ForeignKey('attendee.id'))
+    slug = Column(UnicodeText)
+    comments = Column(UnicodeText, default='')
+
+    __table_args__ = (UniqueConstraint('department_id', 'slug'),)
+
+
+class DeptRole(MagModel):
+    name = Column(UnicodeText)
+    description = Column(UnicodeText)
+    department_id = Column(UUID, ForeignKey('department.id'))
+
+
+class DeptMembership(MagModel):
+    is_dept_head = Column(Boolean, default=False)
+    is_poc = Column(Boolean, default=False)
+    is_checklist_admin = Column(Boolean, default=False)
```

Comment:
I'd originally thought that ``is_poc`` would control whether or not someone was supposed to complete the checklist, but the more I think about it the more I think it's good that you added a separate field for this.  "Point of Contact" sounds more like someone who replies to emails than someone who creates shifts.

With that being said, this means that ``is_poc`` isn't actually doing anything at the moment.  We should probably keep it in, but that definitely makes this more of a speculative column.

Reply:
Haha! You missed the usage in `Attendee.must_contact`. That's the list of people we send to volunteers to contact before an event.